// @react-native-community/cli-platform-ios@12.0.0-alpha.17/build/commands/buildIOS/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _buildProject = require("./buildProject");
var _buildOptions = require("./buildOptions");
var _getConfiguration = require("./getConfiguration");
var _getXcodeProjectAndDir = require("./getXcodeProjectAndDir");
var _pods = _interopRequireDefault(require("../../tools/pods"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

async function buildIOS(_, ctx, args) {
  const {
    xcodeProject,
    sourceDir
  } = (0, _getXcodeProjectAndDir.getXcodeProjectAndDir)(ctx.project.ios);

  // check if pods need to be installed
  await (0, _pods.default)(ctx.root, ctx.dependencies, {
    forceInstall: args.forcePods
  });
  process.chdir(sourceDir);
  const {
    scheme,
    mode
  } = await (0, _getConfiguration.getConfiguration)(xcodeProject, sourceDir, args);
  return (0, _buildProject.buildProject)(xcodeProject, undefined, mode, scheme, args);
}
var _default = {
  name: 'build-ios',
  description: 'builds your app for iOS platform',
  func: buildIOS,
  examples: [{
    desc: 'Build the app for all iOS devices in Release mode',
    cmd: 'npx react-native build-ios --mode "Release"'
  }],
  options: _buildOptions.buildOptions
};
exports.default = _default;
